src/helpers/graph.js

```diff
@@ -0,0 +1,22 @@
+export function addGroupTagToNotes(notes, edges) {
+    if (notes && edges) {
+        notes = notes.map(note => { // add group tag to all the egdes
+            note.group = edges.filter(edge => edge.from === note.id).length > 0 ? 'parent' : 'children' 
+            return note
+        })
+    }
+    return notes
+}
+
+export function removeNoteFromGraph(notes, edges, noteToDelete) {
+    notes = notes.filter(note => note.id !== noteToDelete.id) // remove the note from the array
+    edges = edges.filter(edge => {  // remove all the children of the deleted parent
+      if (edge.from === noteToDelete.id) {
+        notes = notes.filter(note => note.id !== edge.to)
+      } else { return edge }
```

Comment:
I think `filter()` expects a boolean to be returned. Why do you return an object?